vsphere-iso acceptance tests fail due to deprecated language

#### Overview of the Issue

When running acceptance tests for the `vsphere-iso` builder, all of them fail due to deprecated language that needs to be updated.

#### Reproduction Steps

* Per the [contributing](https://github.com/hashicorp/packer/blob/master/.github/CONTRIBUTING.md#opening-an-pull-request) guide

  ```
  PS ~\go\src\github.com\hashicorp\packer> $env:ACC_TEST_BUILDERS='vsphere-iso'
  PS ~\go\src\github.com\hashicorp\packer> $env:ACC_TEST_PROVISIONERS='all'
  PS ~\go\src\github.com\hashicorp\packer> $env:PACKER_ACC=1
  PS ~\go\src\github.com\hashicorp\packer> go test './builder/vsphere/iso/' -v
  ```

### Packer version

```powershell
PS ~\go\src\github.com\hashicorp\packer> .\bin\packer.exe version # Building a feature in #9869
Packer v1.6.3-dev
```

### Simplified Packer Buildfile

N/A

### Operating system and Environment details

```powershell
PS ~\go\src\github.com\hashicorp\packer> systeminfo.exe | Select-String -Pattern '^OS (?:Name|Version)'

OS Name:                   Microsoft Windows Server 2016 Datacenter
OS Version:                10.0.14393 N/A Build 14393
```

### Log Fragments and crash.log files

<details>

```
PS ~\go\src\github.com\hashicorp\packer> $env:ACC_TEST_BUILDERS='vsphere-iso'
PS ~\go\src\github.com\hashicorp\packer> $env:ACC_TEST_PROVISIONERS='all'
PS ~\go\src\github.com\hashicorp\packer> $env:PACKER_ACC=1
PS ~\go\src\github.com\hashicorp\packer> go test './builder/vsphere/iso/' -v
=== RUN   TestISOBuilderAcc_default
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_default (0.04s)
=== RUN   TestISOBuilderAcc_notes
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_notes (0.00s)
=== RUN   TestISOBuilderAcc_hardware
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_hardware (0.00s)
=== RUN   TestISOBuilderAcc_limit
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_limit (0.00s)
=== RUN   TestISOBuilderAcc_sata
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_sata (0.00s)
=== RUN   TestISOBuilderAcc_cdrom
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_cdrom (0.01s)
=== RUN   TestISOBuilderAcc_networkCard
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 2 errors occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
        * Deprecated configuration key: 'network_card'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 2 errors occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
                * Deprecated configuration key: 'network_card'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_networkCard (0.00s)
=== RUN   TestISOBuilderAcc_createFloppy
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_createFloppy (0.01s)
=== RUN   TestISOBuilderAcc_full
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 3 errors occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
        * Deprecated configuration key: 'disk_thin_provisioned'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
        * Deprecated configuration key: 'network_card'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 3 errors occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
                * Deprecated configuration key: 'disk_thin_provisioned'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
                * Deprecated configuration key: 'network_card'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_full (0.01s)
=== RUN   TestISOBuilderAcc_bootOrder
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 3 errors occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
        * Deprecated configuration key: 'disk_thin_provisioned'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
        * Deprecated configuration key: 'network_card'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 3 errors occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
                * Deprecated configuration key: 'disk_thin_provisioned'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
                * Deprecated configuration key: 'network_card'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_bootOrder (0.01s)
=== RUN   TestISOBuilderAcc_cluster
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 1 error occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 1 error occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_cluster (0.00s)
=== RUN   TestISOBuilderAcc_clusterDRS
2020/09/02 18:27:03 [DEBUG] Parsing template...
2020/09/02 18:27:03 [DEBUG] Initializing core...
2020/09/02 18:27:03 [DEBUG] Retrieving 'test' build
2020/09/02 18:27:03 [DEBUG] Preparing 'test' build
2020/09/02 18:27:03 Build 'test' prepare failure: 2 errors occurred:
        * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
        * Deprecated configuration key: 'network'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


    testing.go:146: Prepare error: 2 errors occurred:
                * Deprecated configuration key: 'disk_size'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.
                * Deprecated configuration key: 'network'. Please call `packer fix` against your template to update your template to be compatible with the current version of Packer. Visit https://www.packer.io/docs/commands/fix/ for more detail.


--- FAIL: TestISOBuilderAcc_clusterDRS (0.00s)
=== RUN   TestCreateConfig_Prepare
--- PASS: TestCreateConfig_Prepare (0.00s)
=== RUN   TestStepCreateVM_Run
2020/09/02 18:27:03 ui: Creating VM...
--- PASS: TestStepCreateVM_Run (0.00s)
=== RUN   TestStepCreateVM_RunHalt
2020/09/02 18:27:03 ui: Creating VM...
--- PASS: TestStepCreateVM_RunHalt (0.00s)
=== RUN   TestStepCreateVM_Cleanup
2020/09/02 18:27:03 ui: Destroying VM...
2020/09/02 18:27:03 ui: Destroying VM...
2020/09/02 18:27:03 ui: Destroying VM...
2020/09/02 18:27:03 ui: Destroying VM...
2020/09/02 18:27:03 ui error: destroy failed
--- PASS: TestStepCreateVM_Cleanup (0.00s)
=== RUN   TestStepRemoteUpload_Run
2020/09/02 18:27:03 ui: Uploading path to packer_cache/path
--- PASS: TestStepRemoteUpload_Run (0.00s)
=== RUN   TestStepRemoteUpload_SkipRun
--- PASS: TestStepRemoteUpload_SkipRun (0.00s)
FAIL
FAIL    github.com/hashicorp/packer/builder/vsphere/iso 0.279s
FAIL
```

</details>

--- a/builder/vsphere/iso/builder_acc_test.go
+++ b/builder/vsphere/iso/builder_acc_test.go
@@ -42,8 +42,10 @@ func defaultConfig() map[string]interface{} {
 		"ssh_username": "root",
 		"ssh_password": "jetbrains",
 
-		"vm_name":   commonT.NewVMName(),
-		"disk_size": 2048,
+		"vm_name": commonT.NewVMName(),
+		"storage": map[string]interface{}{
+			"disk_size": 2048,
+		},
 
 		"communicator": "none", // do not start the VM without any bootable devices
 	}
@@ -338,7 +340,9 @@ func TestISOBuilderAcc_networkCard(t *testing.T) {
 
 func networkCardConfig() string {
 	config := defaultConfig()
-	config["network_card"] = "vmxnet3"
+	config["network_adapters"] = map[string]interface{}{
+		"network_card": "vmxnet3",
+	}
 	return commonT.RenderConfig(config)
 }
 
@@ -421,12 +425,18 @@ func fullConfig() map[string]interface{} {
 		"vm_name": commonT.NewVMName(),
 		"host":    "esxi-1.vsphere65.test",
 
-		"RAM":                   512,
-		"disk_controller_type":  "pvscsi",
-		"disk_size":             1024,
-		"disk_thin_provisioned": true,
-		"network_card":          "vmxnet3",
-		"guest_os_type":         "other3xLinux64Guest",
+		"RAM": 512,
+		"disk_controller_type": []string{
+			"pvscsi",
+		},
+		"storage": map[string]interface{}{
+			"disk_size":             1024,
+			"disk_thin_provisioned": true,
+		},
+		"network_adapters": map[string]interface{}{
+			"network_card": "vmxnet3",
+		},
+		"guest_os_type": "other3xLinux64Guest",
 
 		"iso_paths": []string{
 			"[datastore1] ISO/alpine-standard-3.8.2-x86_64.iso",
@@ -550,7 +560,9 @@ func clusterDRSConfig() string {
 	config["cluster"] = "cluster2"
 	config["host"] = ""
 	config["datastore"] = "datastore3" // bug #183
-	config["network"] = "VM Network"   // bug #183
+	config["network_adapters"] = map[string]interface{}{
+		"network": "VM Network",
+	}
 
 	return commonT.RenderConfig(config)
 }

